pillar/vector/mitx.sls

```diff
@@ -0,0 +1,371 @@
+        inputs:
+          - worker_lms_stderr_log
+        type: regex_parser
+        field: message
+        overwrite_target: true
+        patterns:
+          - '(?P<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3}) (?P<log_level>[A-Z]+) (?P<pid>\d+) \[(?P<module>.*?)\] \[user (?P<user>.*?)\] \[ip (?P<client_ip>.*?)\] (?P<code_loc>.*?) - (?P<message>.*)'
+        types:
+          time: timestamp|%Y-%m-%d %H:%M:%S,%3f
+
+      worker_stderr_log_labeler:
+        inputs:
+          - worker_cms_stderr_log_parser
+          - worker_lms_stderr_log_parser
+        type: add_fields
+        fields:
+          labels:
+            - edx_worker
+
+      worker_stderr_timestamp_renamer:
+        inputs:
+          - worker_stderr_log_labeler
+        type: rename_fields
+        fields:
+          time: "@timestamp"
+
+      {% endif %}
+
+      tracking_log_parser:
+        inputs:
+          - tracking_log
+        type: json_parser
+        field: message
+        drop_field: true
+
+      tracking_log_timestamp_coercer:
+        inputs:
+          - tracking_log_parser
+        type: coercer
+        types:
+          time: timestamp|%Y-%m-%dT%H:%M:%S%.6f%:z
+
+      tracking_log_timestamp_renamer:
+        inputs:
+          - tracking_log_timestamp_coercer
+        type: rename_fields
+        fields:
+          time: "@timestamp"
+
+      auth_log_parser:
+        inputs:
+          - auth_log
+        type: regex_parser
+        field: message
+        overwrite_target: true
+        patterns:
+          - '^(?P<time>\w{3} \d{2} \d{2}:\d{2}:\d{2}) \S+ (?P<service>.*?)\[(?P<pid>\d+)\]: (?P<message>.*)'
+        fields:
+          time: timestamp|%b %d %H:%M:%S
+
+      auth_log_sampler:
+        inputs:
+          - auth_log_parser
+        type: sampler
+        rate: 1
+        exclude:
+          type: check_fields
+          "message.contains": "CRON"
+
+    sinks:
+
+      {% if 'edx' in salt.grains.get('roles') %}
+
+      elasticsearch_nginx_access:
+        inputs:
+          - nginx_access_log_timestamp_renamer
+        type: elasticsearch
+        endpoint: 'http://operations-elasticsearch.query.consul:9200'
+        index: mitx-nginx-access-%Y.%W
+        healthcheck: false
+
+      elasticsearch_nginx_error:
+        inputs:
+          - nginx_error_log_timestamp_renamer
+        type: elasticsearch
+        endpoint: 'http://operations-elasticsearch.query.consul:9200'
+        index: mitx-nginx-error-%Y.%W
+        healthcheck: false
+
+      {% endif %}
+
+      elasticsearch_lms_cms_worker:
+        inputs:
+          {% if 'edx' in salt.grains.get('roles') %}
+          - cms_stderr_sampler
+          - lms_stderr_sampler
+          {% endif %}
+          {% if 'edx-worker' in salt.grains.get('roles') %}
+          - worker_stderr_timestamp_renamer
+          {% endif %}
+        type: elasticsearch
+        endpoint: 'http://operations-elasticsearch.query.consul:9200'
+        index: mitx-%Y.%W
```

Comment:
I feel that this index name doesn't match the naming convention used for the other logs. Might want to either have multiple indices or if we're gonna use one index for all those logs, maybe use some more descriptive for it `mitx-stderr...`

Reply:
I agree; adding `stderr` would make its nature more clear.